src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -695,21 +694,34 @@ export class StudyViewPageStore {
     });
 
     // includes all physical studies from queried virtual studies
-    readonly queriedPhysicalStudyIds = remoteData({
-        await: ()=>[this.filteredPhysicalStudies, this.filteredVirtualStudies],
+    readonly queriedPhysicalStudies = remoteData({
+        await: () => [this.filteredPhysicalStudies, this.filteredVirtualStudies],
         invoke: async () => {
-            let studyIds = _.reduce(this.filteredPhysicalStudies.result, (acc, next) => {
-                acc[next.studyId] = true;
+
+            const physicalStudiesSet = this.physicalStudiesSet.result;
+
+            let studies = _.reduce(this.filteredPhysicalStudies.result, (acc, next) => {
+                acc[next.studyId] = physicalStudiesSet[next.studyId];
                 return acc;
-            }, {} as { [id: string]: boolean })
+            }, {} as { [id: string]: CancerStudy })
 
             this.filteredVirtualStudies.result.forEach(virtualStudy => {
                 virtualStudy.data.studies.forEach(study => {
-                    studyIds[study.id] = true;
+                    if (!studies[study.id]) {
+                        studies[study.id] = physicalStudiesSet[study.id];
+                    }
                 })
             });
+            return _.values(studies);
+        },
+        default: []
+    });
 
-            return Object.keys(studyIds);
+    // includes all physical studies from queried virtual studies
+    readonly queriedPhysicalStudyIds = remoteData({
+        await: () => [this.queriedPhysicalStudies],
+        invoke: async () => {
```

Comment:
when we a mobxpromise is just a "facade" for another mobxpromise, i.e. there's no actualy async operation inside invoke, we should take asycn tag of method and return Promise.resolve(data).  A little clearer.